### PR TITLE
fix: reload default page on discard

### LINKED ISSUE #13736: [Bug]: Discard the change staying in the newly created page throws a page not found error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Discard the change staying in the newly created page throws a page not found an error

### Steps To Reproduce

1. Connect the application to git 
2. now add a new page 
3. discard the change
4. The user will be navigated to page not found error

### Public Sample App

_No response_

### Version

release

### PR DESCRIPTION
**Pull request by IAmAnubhavSaini**
## Description

After discard is successful, reload the default page.

Fixes #13736

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by IAmAnubhavSaini** [comment]
## Loading default page after discard.

### bug

https://user-images.githubusercontent.com/1573771/177031576-e934b48f-7988-42c6-a475-8042a7fa479f.mov

## fix

https://user-images.githubusercontent.com/1573771/177031585-6700327a-9002-4d9e-9889-e4fd3625c662.mov

**Comment by IAmAnubhavSaini** [comment]
Failing tests. This needs update in discard tests.

Screenshot 2022-07-03 at 3 34 23 PM

**Comment by IAmAnubhavSaini** [comment]
Failing tests; these might be genuine failures due to code change:

Screenshot 2022-07-04 at 8 51 47 AM

**Comment by IAmAnubhavSaini** [comment]
Only one test fails after fixing another one. Who knew that fixing failing tests works?

Screenshot 2022-07-04 at 11 41 48 AM

**Comment by IAmAnubhavSaini** [comment]
Failing build

Screenshot 2022-07-05 at 11 05 40 AM
Screenshot 2022-07-05 at 11 04 57 AM
Screenshot 2022-07-05 at 11 04 52 AM

**Comment by IAmAnubhavSaini** [comment]
Failing tests

Screenshot 2022-07-06 at 11 46 19 AM

**Comment by IAmAnubhavSaini** [comment]
Failing test, cannot run it locally.

**Comment by IAmAnubhavSaini** [comment]
Failing tests

Screenshot 2022-07-12 at 11 43 00 AM

**Comment by IAmAnubhavSaini** [comment]
Failing test.
Screenshot 2022-07-19 at 6 09 23 PM

**Comment by IAmAnubhavSaini** [comment]
Tests are passing.
Screenshot 2022-07-19 at 7 13 17 PM

File: app/client/src/sagas/GitSyncSagas.ts [hunk 1]
```diff
@@ -858,9 +858,14 @@ function* discardChanges() {
       getLogToSentryFromResponse(response),
     );
     if (isValidResponse) {
-      yield put(discardChangesSuccess(response?.data));
+      yield put(discardChangesSuccess(response.data));
+      // const applicationId: string = response.data.id;
+      const pageId: string = response.data.pages.filter(
```
Comment (berzerkeer) [inline app/client/src/sagas/GitSyncSagas.ts:863]: Any specific reason as to why using `filter` over `find` here ?

Comment (IAmAnubhavSaini) [reply app/client/src/sagas/GitSyncSagas.ts:863]: No particular reason. Will update in some future pr to `find`. If this is not a blocker please approve.

**Review by berzerkeer — APPROVED** [review]

**Comment by RakshaKShetty** [comment]
@IAmAnubhavSaini  this looks gud

**Pull request merged**